src/actions/notifications.rs

```diff
@@ -150,18 +149,19 @@ impl BlockingNotificationAction for Cancel {
 
 impl BlockingNotificationAction for DidChangeConfiguration {
     fn handle<O: Output>(
-        mut params: DidChangeConfigurationParams,
+        params: DidChangeConfigurationParams,
         ctx: &mut InitActionContext,
         out: O,
     ) -> Result<(), ()> {
         trace!("config change: {:?}", params.settings);
-	params.settings = crate::lsp_data::json_key_to_snake_case(params.settings);
-        let settings = ChangeConfigSettings::deserialize(&params.settings);
+        let mut dups = std::collections::HashMap::<String, Vec<String>>::new();
+        let mut unknowns = Vec::<String>::new();
```

Comment:
I think inference is preferable here, with some importing.
```rust
let mut dups = HashMap::new();
let mut unknowns = Vec::new();
```

More importantly these duplicates and unknowns are not actually used.

Reply:
Ok